extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/Endpoints.java

```diff
@@ -220,7 +220,12 @@ public void handle(Void event) {
             timerId = vertx.setPeriodic(autoPingInterval.get().toMillis(), new Handler<Long>() {
                 @Override
                 public void handle(Long timerId) {
-                    connection.sendAutoPing();
+                    if (connection.isOpen()) {
+                        connection.sendAutoPing();
+                    } else {
+                        LOG.debugf("Try to cancel the autoPing timer for a closed connection: %s", connection.id());
+                        vertx.cancelTimer(timerId);
```

Comment:
Is it ok to cancel the timer from within the callback? @vietj

Reply:
I think this is important as my understanding is that you will have the connection closed before executing the code below that cancels the timer?

Or am I mistaken?

Reply:
FWIW we (or at least I) did it in some OIDC stuff and we didn't detect issues.

Reply:
> I think this is important as my understanding is that you will have the connection closed before executing the code below that cancels the timer?
> 
> Or am I mistaken?

Yes, when the timer is invoked and the connection is closed then we cancel timer completely.

Reply:
timers can be canceled any time from any thread